pkg/controller/backup_controller_test.go

```diff
@@ -260,14 +262,14 @@ func TestProcessBackup(t *testing.T) {
 
 			expectedStatusKeys := 2
 			if test.backup.Spec.TTL.Duration > 0 {
-				assert.True(t, collections.HasKeyAndVal(patch, "status.expiration", expiration.UTC().Format(time.RFC3339)), "patch's status.expiration does not match")
+				assert.True(t, hasKeyAndVal(patch, expiration.UTC().Format(time.RFC3339), "status", "expiration"), "patch's status.expiration does not match")
 				expectedStatusKeys = 3
 			}
 
-			assert.True(t, collections.HasKeyAndVal(patch, "status.version", float64(1)))
-			assert.True(t, collections.HasKeyAndVal(patch, "status.phase", string(v1.BackupPhaseInProgress)), "patch's status.phase does not match")
+			assert.True(t, hasKeyAndVal(patch, float64(1), "status", "version"))
```

Comment:
I know this isn't a change you're making as part of this PR, but I'm curious why this is `float64` and not `int`, since that's what backup.status.version is?

Reply:
This is because `patch` is a `map[string]interface{}` being created from a `json.Unmarshal` call which uses `float64` for numbers.

Reply:
thx